bridge/src/commonMain/kotlin/ru/herobrine1st/matrix/bridge/telegram/TelegramWorker.kt

```diff
@@ -186,17 +190,70 @@ class TelegramWorker(
                     if (result.value.isEmpty()) continue // Short-circuit empty responses
                     result.value.forEach { update ->
                         update.message?.let { message ->
+                            // will likely throw exceptions, but docs say it is null only in channels
                             val (chat, sender, content) = run {
                                 message.text?.let { text ->
-                                    // will likely throw exceptions, but docs say it is null only in channels
                                     val sender = message.from!!
                                     return@run Triple(
                                         message.chat,
                                         sender,
                                         RoomMessageEventContent.TextBased.Text(body = text),
                                     )
                                 }
-                                return@forEach // TODO respond with unhandled event
+
+                                message.photo?.maxByOrNull { it.fileSize ?: 0 }?.let { photo ->
+
+                                    val sender = message.from!!
+
+                                    val fileInfo = withContext(Dispatchers.IO) {
+                                        bot.getFile(photo.fileId).toResult().get()
+                                    }
+
+                                    val caption = message.caption ?: "none"
+                                    val responseBody =
+                                        fileInfo.filePath?.let { filePath ->
+                                            bot.downloadFile(filePath).toResult().get()
+                                        } ?: return@run Triple(
+                                            message.chat,
+                                            sender,
+                                            RoomMessageEventContent.TextBased.Notice(
+                                                body = "Could not download photo. Caption was: $caption",
+                                            ),
+                                        )
+
+                                    val mxcUrl = withContext(Dispatchers.IO) {
+                                        client.media.upload(
+                                            Media(
+                                                content = responseBody.byteStream().toByteReadChannel(),
+                                                contentLength = responseBody.contentLength(),
+                                                contentType = responseBody.contentType()?.let {
+                                                    ContentType(it.type(), it.subtype())
+                                                },
+                                                contentDisposition = ContentDisposition(disposition = "image.jpg"),
```

Comment:
And actually it is not simply a filename
```suggestion
                                                contentDisposition = ContentDisposition.Attachment.withParameter("filename", "image.jpg"),
```